### ISSUE TITLE
Three windows next to each other when opening a new window

### ISSUE THREAD

**Issue by zjeffer** [baskerville/bspwm#876]
If I have two windows next to each other, and one of them is even slightly larger than the other (like in situation X), I run into the following problem:

```
 +-----------------------+ 
 |         |             | 
 |         |             |
 |         |             | 
 |    1    |      2      |
 |         |             |  
 |         |             | 
 |         |             |  
 +-----------------------+ 

             X                        
```

If I open a new window while window 2 is selected, it will create 3 windows next to each other (Y), instead of what I'd like to happen (Z)

```
 +-----------------------+ 
 |         |      |      |
 |         |      |      |
 |         |      |      |
 |    1    |  2   |   3  |
 |         |      |      |
 |         |      |      |
 |         |      |      |
 +-----------------------+ 

              Y
```

```

+-----------------------+
|         |             |
|         |      2      |    
|         |             |     
|    1    |-------------|
|         |             |       
|         |      3      |    
|         |             |    
+-----------------------+ 

             Z
```

I don't know if this is because of the newest update or if this has always been this way, but today was the first time I noticed this.

EDIT: I basically NEVER want to have 3 windows next to each other, if that's possible.

**Comment by baskerville** [comment]
The way new windows are inserted is explained in the README.

**Comment by zjeffer** [comment]
I tried both `longest_side` and `spiral` and they both do the wrong thing. Did you mean a different setting?

**Comment by zjeffer** [comment]
Any update on this? This is really bugging me :/

**Comment by cjk** [comment]
@zjeffer When using the default (spiral-) layout, make sure you split in the correct place. For me, when I split in the smaller window, it gets split horizontally. But when I split in the larger of the two initial windows, it get's split vertically.
So, you have to focus the correct window in order to get what you want.

**Comment by zjeffer** [comment]
@cjk The spiral layout is not what I want at all, I really can't use that layout. With longest_side, focusing on the larger window always splits vertically.

**Comment by baskerville** [comment]
> When the focused window is larger than all the other windows, it will always split vertically. When it's smaller than (or as large as) all the other windows, it will split horizontally.

The `longest_side` automatic insertion scheme, as its name suggests, splits windows along the longest side.

> I'd like it to split horizontally no matter what.

You could have:
```sh
bspc query -N -n focused.automatic && echo "split_dir=south"
```
in your external rules command but I doubt that's what you're after.

Could you provide a rigorous description of what you're trying to achieve?

**Comment by zjeffer** [comment]
Sure, but it's hard to explain:

When only two windows are present on the whole desktop, they should always be next to each other. 
When three windows are present, one of them should be vertical and the other two on top of each other, like in situation Z. Layout Y should never be possible, because I think that 3 vertical windows are not very space efficient. 

Here's an image of what I mean: https://imgur.com/lzJ8wzr
Basically, the node tree on the left should always result in the layout in the center, never the one on the right, even when (like in situation X) the focused window is larger. What I want is 'true binary', where it never looks like a tree node has 3 leaves.

Also, like described here: https://github.com/baskerville/bspwm/issues/872#issuecomment-439682363, `bspc config removal_adjustment longest_side` still creates 3 vertical windows when the left window is smaller. I created another video here: https://gfycat.com/InsistentEasyAllensbigearedbat
First, the behaviour when closing one of the bottom right windows is correct and just how I want it. Then, I increase the size of the bottom right windows, and when I then close one of them, it again puts all 3 windows in a vertical layout, rather than two of them horizontal. 

In that same issue thread, I noticed that both behaviours of #872 still occur when moving a window to a different monitor or workspace.

All of this would be fixed if it was impossible to have vertical 3 windows next to each other. Sorry for the wall of text :)
